src/math/fibonacci.lua

```diff
@@ -0,0 +1,80 @@
+-- Fibonacci.lua
+
+-- Fibonacci Sequence:
+-- The Fibonacci sequence is a series of numbers where each number (after the first two) is the sum of the two preceding ones.
+-- More information: https://en.wikipedia.org/wiki/Fibonacci_number
+
+-- Author: Gyandeep
+
+--      Implemented nth Fibonacci number using different approaches in O(2^N) , O(N) , O(logN) , O(1) .
+
+
+-- Function to calculate Fibonacci numbers using recursion
+-- Time Complexity: O(2^n)
+function fibonacci_recursive(n)
+    if n <= 0 then
+        return 0
+    elseif n == 1 then
```

Comment:
I prefer omitting the redundant `else` in `if`-`return` trees in Lua, so I would rewrite this as

```lua
-- or more concise: `if n <= 1 then return n end`
if n <= 0 then return 0 end
if n == 1 then return 1 end
return fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2)
```